pills/04-basics-of-language.xml

```diff
@@ -4,5 +4,362 @@
+  </para>
+
+  <para>
+    <emphasis role="underline">Important</emphasis>: in Nix, everything is an
+    expression, there are no statements. This is common to many functional
+    languages.
+  </para>
+
+  <para>
+    <emphasis role="underline">Important</emphasis>: values in Nix are
+    immutable.
+  </para>
+
+  <section>
+    <title>Value types</title>
+
+    <para>
+      We've installed nix-repl in the previous pill. If you didn't,
+      <literal>nix-env -i nix-repl</literal>. The nix-repl syntax is slightly
+      different than nix syntax when it comes to assigning variables, but no
+      worries.  I prefer playing with nix-repl with you before cluttering your
+      mind with more complex expressions.
+    </para>
+
+    <para>
+      Launch nix-repl. First of all, nix supports basic arithmetic operations:
+      +, -, and *. The integer division can be done with builtins.div.
+    </para>
+
+    <screen><xi:include href="./04/basics.txt" parse="text" /></screen>
+
+    <para>
+      Really, why doesn't nix have basic operations such as division? Because
+      it's not needed for creating packages. Nix is not a general purpose
+      language, it's a domain-specific language for writing packages.
+    </para>
+
+    <para>
+      Just think that builtins.div is not being used in the whole of our
+      nixpkgs repository: it's useless.
+    </para>
+
+    <para>
+      Other operators are ||, &amp;&amp; and ! for booleans, and relational
+      operators such as !=, ==, &lt;, >, &lt;=, >=. In Nix, &lt;, >, &lt;= and
+      >= are not much used. There are also other operators we will see in the
+      course of this series.
+    </para>
+
+    <para>
+      Nix has integer (not floating point), string, path, boolean and null
+      <link xlink:href="http://nixos.org/nix/manual/#ssec-values">simple</link>
+      types. Then there are lists, sets and functions. These types are enough
+      to build an operating system.
+    </para>
+
+    <para>
+      Nix is strongly typed, but it's not statically typed. That is, you
+      cannot mix strings and integers, you must first do the conversion.
+    </para>
+
+    <para>
+      Try to use <literal>/</literal> between two numbers:
+    </para>
+
+    <screen><xi:include href="./04/relative-path.txt" parse="text" /></screen>
+
+    <para>
+      Nix parsed 2/3 as a relative path to the current directory. Paths are
+      parsed as long as there's a slash. Therefore to specify the current
+      directory, use <literal>./.</literal> In addition, Nix also parses urls.
+    </para>
+
+    <para>
+      Not all urls or paths can be parsed this way. If a syntax error occurs,
+      it's still possible to fallback to plain strings. Parsing urls and paths
+      are convenient for additional safety.
+    </para>
+  </section>
+
+  <section>
+    <title>Identifier</title>
+
+    <para>
+      Not much to say, except that dash (-) is allowed in identifiers. That's
```

Comment:
maybe a good spot for a `<code>` (or some other tag, but not sure which one.) and several other places here could be improved with a `<code>`.